lib/spack/spack/spec.py

```diff
@@ -1016,7 +1140,9 @@ def __init__(self, spec, name, query_parameters):
         super(SpecBuildInterface, self).__init__(spec)
         # Adding new attributes goes after super() call since the ObjectWrapper
         # resets __dict__ to behave like the passed object
-        self.token = spec, name, query_parameters
+        original_spec = getattr(spec, 'wrapped_obj', spec)
+        self.wrapped_obj = original_spec
+        self.token = original_spec, name, query_parameters
```

Comment:
Should this logic be pushed into lang.ObjectWrapper? It seems like it would apply to any object that cares about it's id, so we could use `__getattr__` and `__getitem__` to make sure methods are called on the wrapped object.

Reply:
Hmm, I need to think about that. I think `__getattr__` wouldn't work since the way we create a wrapper object is to fake the class name and share the `__dict__` object:

https://github.com/spack/spack/blob/c791ddc742474226bed94fcfadd5965148957918/lib/spack/llnl/util/lang.py#L742-L747

We can probably create though a better API to say that a method should be forwarded to the real object.